### DISCUSSION TITLE
Add ability to disable or customize highlighting in loki query results

### DISCUSSION THREAD

**Discussion by na--** [grafana/grafana#52044] [Observability]
I noticed a potential UX issue in the Grafana interface for displaying loki logs :disappointed: There doesn't seem to be a way to disable or change the orange highlighting of the log parts that match the query filters, right? Because I think that, in the majority of cases, I would want to do that...

The current approach is a bit weird to me. Why would I always want to highlight the only part of the logs that I know exists on _every_ line? :sweat_smile: If anything, it is distracting me from the rest of the information, which is most often the part I am interested in. If possible, I'd even sometimes want to do the reverse and un-highlight the parts I have matched, to de-emphasize them... For example, it'd be nice if I can use a greyer font color (and no highlighting) for the matched parts when I filter by a request ID or trace ID or some ID like that.

One use case of the current orange highlight mode I can think of is when I am actually searching for information I don't know, e.g. usually with some sort of a regular expression or something like that. Or to bring attention to a value that is adjacent to the highlighed match by focusing my eyes next to it. But when I am filtering based on information that I know, to get a specific sub-set of logs with the intention of sequentially reading through them to diagnose an issue, which is my most frequent use case for logs, the bright orange highlight of already known information only distracts me.

My suggestion would be to de-couple the highlighting logic from the search/filter expressions and make it configurable. For example, if I have the following filter expression `{environment="production"} |= "foo" |= "bar"`, after I run the query, ideally I should be able to disable or change the highlight and text color for any of the `foo` and `bar` filters separately :thinking: This is backwards compatible, but much more flexible - _by default_, all of the highlights will be the same as they are now, but we'd be able to edit or disable them to suit our needs. 

It'd also be nice to be able to highlight matches in the logs without actually filtering the results to just the log entries that contain these highlights. In the above example, I might want to highlight the word `baz` in the search results and share the link with a colleague, to bring attention to a few specific log entries out of the subset that contains `foo` and `bar`, without actually removing the context by limiting the results to just logs that contain all three words.

**Comment by RutgerKe** [comment]
Ran into the same issue: Filtering down a log stream (for example `|= "error"`), displaying them in a panel (calling the panel 'application errors') and getting the distracting highlighting on the only part of the log line that is the same in every line (`error`, which is already present in the title), and thus the part that is not interesting to look at. (distracting from actual error content).

A toggle to turn this on/off would already be a big help here.

**Comment by henry-swatchon** [comment]
I was also looking for the same feature to disable the highlights and found a little trick.
Set a variable to the filter, in my case `|= "error"` and set the filter to that variable. 
(so the variable `error` is string `error`. 

`{namespace="$namespace", pod=~"$pod"} |~ "$error"` does to trick instead of 
`{namespace="$namespace", pod=~"$pod"} |~ "error"`

**Reply by misinformed329** [reply]
Thank you so much for that. The highlighting was driving me insane!

**Reply by ScreNN** [reply]
Can you tell me how to remove the backlight? It didn't work for me.

I have this request
`{container_name="traefik"} | logfmt |= "request denied"`

**Comment by cedsana** [comment]
This is annoying, I'm always using `line_format` to omit the part of the logs that get highlighted, it's very very distracting.

**Comment by pbolcar** [comment]
Any chance this has been handled elsewhere? I am having the same issue.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was previously converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/68572).
